Index: application/configs/config.ini
===================================================================
--- application/configs/config.ini	(revision 820)
+++ application/configs/config.ini	(working copy)
 
 ; Display Information
 ; Header Infomation
@@ -53,6 +53,14 @@
 ; By default it is cached.
 SOAP.cacheWSDL = false
 
+;Database for user password information
+db.adapter = PDO_MYSQL
+db.params.host = localhost
+db.params.username = authuser
+db.params.password = auth1
+db.params.dbname = sahara
+db.params.port = 3306
+
 [development : production]
 
 logger.level = DEBUG
Index: application/controllers/IndexController.php
===================================================================
--- application/controllers/IndexController.php	(revision 820)
+++ application/controllers/IndexController.php	(working copy)
@@ -55,6 +55,11 @@
         $form = new Sahara_Auth_Form();
         $this->view->form = $form;
 
+        // Configure database and store to the registry.
+        $db = Zend_Db::factory($this->_config->db);
+        Zend_Db_Table_Abstract::setDefaultAdapter($db);
+        Zend_Registry::set('db', $db);
+
         if ($this->_request->isPost() && $form->isValid($this->_request->getParams()))
         {
             Zend_Session::regenerateId();
@@ -65,14 +70,30 @@
             /******************************************************************
              ** TODO Add your authentication.                                **
              ******************************************************************/
-
+            $adapter = new Zend_Auth_Adapter_DbTable(Zend_Registry::get('db'),
+									'users',		// Contact table
+									'name', 		// Column containing identity
+									'password', // Column containing credential
+									'SHA1(?)');
+            
+			$adapter->setIdentity($username);
+			$adapter->setCredential($password);
+			$result = $this->_auth->authenticate($adapter);
+			
+			if ($result->getCode() != Zend_Auth_Result::SUCCESS)
+			{
+				/* Failed authentication, so redisplay form with error message. */
+				array_push($this->view->messages, "Authentication has failed. Please enter user name and password.");
+				$this->view->form->populate($this->_request->getParams());
+				return;				
+			}
+            
             /* Store the authentication information. */
             $qName =  $inst . ':' . $username;
             $storage = $this->_auth->getStorage();
             $storage->clear();
             $storage->write($qName);
 
-
             $user = Sahara_Soap::getSchedServerPermissionsClient()->getUser(array('userQName' => $qName));
             if ($user->persona == 'NOTFOUND')
             {
@@ -81,22 +102,22 @@
                  **************************************************************/
             }
 
-            /* Redirect to an appropriate page. */
-            switch ($user->persona)
-            {
-                case Sahara_Acl::USER:
-                    $this->_redirectTo('index', 'queue');
-                    break;
-                case Sahara_Acl::ACADEMIC:
-                    $this->_redirectTo('index', 'academic');
-                    break;
-                case Sahara_Acl::ADMIN:
-                    $this->_redirectTo('index', 'admin');
-                    break;
-                default:
-                    $this->view->messages = array("Unknown user \"$qName\".");
-                    break;
-            }
+           /* Redirect to an appropriate page. */
+		   switch ($user->persona)
+		   {
+		   		case Sahara_Acl::USER:
+		        	$this->_redirectTo('index', 'queue');
+		            break;
+		        case Sahara_Acl::ACADEMIC:
+		        	$this->_redirectTo('index', 'academic');
+		            break;
+		        case Sahara_Acl::ADMIN:
+		         	$this->_redirectTo('index', 'admin');
+		            break;
+		       default:
+		            $this->view->messages = array("Unknown user \"$qName\".");
+		            break;
+		     }
         }
     }
 
